### DISCUSSION TITLE
Enabling debug logs for TFTP?

### DISCUSSION THREAD

**Discussion by BloodBlight** [antonym/netboot.xyz#1133] [General] [answered]
I am having an odd issue where one system boots to the main menu just fine, but another reports that the file was not found.

Unfortunately there is no logging of who accessed what files.  Seems like this should be a default on (especially when a client attempts to DL a file that is not there), but is there a way to enable verbose logging for the TFPT server?

I am actually using the linuxserver/netbootxyz docker image, but this seems like it should just be a variable that could be set.

Thanks for any feedback.  :)

**Comment by antonym** [comment]
Have you tried:
```
docker logs -f netbootxyz
2022-07-08 01:11:17 info in.tftpd[110]: sending NAK (1, File not found) to 10.0.100.139
2022-07-08 01:11:17 notice in.tftpd[111]: RRQ from 10.0.100.139 filename menu.ipxe
2022-07-08 01:11:17 notice in.tftpd[112]: RRQ from 10.0.100.139 filename boot.cfg
2022-07-08 01:11:20 notice in.tftpd[113]: RRQ from 10.0.100.139 filename linux.ipxe
```

**Comment by BloodBlight** [comment, marked as answer]
Hmmmm, ya, I did.  Even booting several systems successfully didn't return any events.

But is not the "official" docker though, so maybe something bugged in their build?

**Reply by antonym** [reply]
Give our official one a try from https://github.com/netbootxyz/docker-netbootxyz:

```
docker pull ghcr.io/netbootxyz/netbootxyz
```

If there's anything we could add to it, let us know. I don't believe the linuxserver one logs tftpd.

**Comment by BloodBlight** [comment]
Yep, that did it!  Now I just have to figure out why my client is adding extra characters to the file name...   /eyeroll
```
2022-07-08 15:28:56 notice in.tftpd[88]: RRQ from 192.168.10.206 filename netboot.xyz.kpxe�����
```

Thanks!

**Reply by BloodBlight** [reply]
Seems to be this char if anyone is interested:
https://www.fileformat.info/info/unicode/char/fffd/index.htm

**Comment by exptom** [comment]
@BloodBlight Did you manage to solve this? I have a similar problem with a dell laptop appending extra characters to the filename!

`2023-02-13 12:34:08 notice in.tftpd[68]: RRQ from 10.10.10.53 filename netboot.xyz.efi.���`

**Reply by BloodBlight** [reply]
Nope, but I opened a bug report just the other day for it:
https://github.com/netbootxyz/netboot.xyz/issues/1198

I also took a second to confirm that what you are seeing is the same U+FFFD Unicode.  So you might want to drop in and say you are also seeing the issue.  Probably want to list the exact hardware you are using.
